[Bryan Chan Kah Hoe} iP

bryanckh opened the pull request:
BeeChat

“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

BeeChat frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST - **SUPER FAST** to use

Here is how to show all tasks

`list`

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)
 
If you are a Java programmer, you can use it to practice Java too. Here's the main method:

`public class Main {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

bryanckh committed:
Level 0 - Rename, Greet, Exit

bryanckh committed:
Level 1 - Echo

bryanckh committed:
Level 2 - Add, List

bryanckh committed:
Level 3 - Mark as Done

bryanckh committed:
Level 4 - ToDos, Events, Deadlines

bryanckh committed:
Level 5 - Handle Errors

bryanckh committed:
Level 6 - Delete

bryanckh committed:
A-TextUiTesting

LynetteLeeSiYing commented on src/main/java/Beechat.java:
```diff
@@ -0,0 +1,104 @@
+                msg = sc.nextLine();
+                continue;
+            }
+            if (start.equals("todo")) {
+                try {
+                    String description = (msg.split("todo "))[1];
+                    task = new TodoTask(description);
+                    list.add(task);
+                    System.out.println("added: " + description);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Please enter the todo task in this form: \n" +
+                            "todo [task name]");
+                }
+            }
+            if (start.equals("deadline")) {
+                try {
+                    String description = ((msg.split("deadline "))[1]).split("/by ")[0];
+                    String by = (msg.split("/by "))[1];
+                    task = new DeadlineTask(description, by);
+                    list.add(task);
+                    System.out.println("added: " + description);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Please enter the deadline task in this form: \n" +
+                            "deadline [task name] /by [deadline]");
+                }
+            }
+            if (start.equals("event")) {
+                try {
+                    String description = ((msg.split("event "))[1]).split("/from ")[0];
+                    String from = (msg.split("/from "))[1].split("/to ")[0];
+                    String to = (msg.split("/to "))[1];
+                    task = new EventTask(description, from, to);
+                    list.add(task);
+                    System.out.println("added: " + description);
```
Since there are several repeated blocks to deal with each command, perhaps you can considering create another method specially to handle the similar task, e.g. Parser, to improve code readability

LynetteLeeSiYing commented on src/main/java/Beechat.java:
Using more OOP is encouraged, to organise your code efficiently, and easier to maintain or scale up any features in the future

LynetteLeeSiYing commented on src/main/java/Task.java:
I noticed that most of the methods are missing JavaDoc comments. It would be great to add them to improve clarity and maintainability.

LynetteLeeSiYing commented on src/main/java/Beechat.java:
There some some coding standard violations here, mainly in naming of your variable in this case, e.g. 'list' would be recommended to be 'tasks' as plural noun of task indicating what is the list is containing.

LynetteLeeSiYing commented on src/main/java/Beechat.java:
Could you manage to load and store your history tasks in somewhere if there is no 'Storage' class handling it in your code?

LynetteLeeSiYing reviewed (commented):
LGTM. Just some few nits to fix :)

bryanckh committed:
Level 7 - Save

bryanckh committed:
Level 7 - Save p2

bryanckh committed:
Merge branch 'branch-Level-7'

bryanckh committed:
Level 8 - Dates and Times

bryanckh committed:
Merge branch 'branch-level-8'

bryanckh committed:
Level 8 part 2 - A-MoreOOP

bryanckh committed:
Merge branch 'branch-A-MoreOOP'

bryanckh committed:
Level 8 part 3 - A-Packages

bryanckh committed:
Merge branch 'branch-A-Packages'

bryanckh committed:
Merge branch 'add-gradle-support'

bryanckh committed:
Level 8 part 4 - A-JUnit

bryanckh committed:
Level 8 part 4 - A-Jar

bryanckh committed:
Level 9 part 1 - A-JavaDoc

bryanckh committed:
Level 9 part 2 - A-CodingStandard

bryanckh committed:
Level 9 - Find

bryanckh committed:
Merge branch 'branch-A-JavaDoc'

bryanckh committed:
Merge branch 'branch-A-CodingStandard'

bryanckh committed:
Merge branch 'branch-Level-9'

bryanckh committed:
Level 10 - GUI

bryanckh committed:
Merge branch 'branch-Level-10'

bryanckh committed:
Add assertions to Storage and AddCommand classes

The addition of Assertions to the Storage and AddCommand classes ensure that
critical assumptions hold true during runtime.

Currently, the Storage class performs file operations without checking if the files exist or the tasks are present.
Similarly, the AddCommand class does not check the user input sufficiently if there is a description.

Adding assertions help to ensure tasks and files are non-null before saving/loading,
and also ensure task descriptions are provided in AddCommand

These assertions will also help to catch potential logic errors during the
execution of the application.

bryanckh committed:
Add code to improve code quality, A-CodeQuality

bryanckh committed:
Merge pull request #2 from bryanckh/branch-A-Assertions

Add assertions to Storage and AddCommand classes

bryanckh committed:
Merge pull request #3 from bryanckh/branch-A-CodeQuality

Add code to improve code quality, A-CodeQuality

bryanckh committed:
Add C-Help Extension

bryanckh committed:
Merge branch 'branch-BCD-Extension'

bryanckh committed:
Add User Guide

bryanckh committed:
Add  code and GUI fixes

bryanckh committed:
Add GUI fixes
